OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
+
 private:
-    
+    /// Compute Transform of this geometry relative to its base frame, utilizing 
+    /// passed in state. Both transform and body_id are set in the passed-in 
+    /// decorations as a side effect.
+    void setDecorativeGeometryTransform(
+        SimTK::Array_<SimTK::DecorativeGeometry>& decorations,
+        const SimTK::State& state) const;
+
+    /// Manage Appearance (how the Geometry is rendered) by applying Appearance 
+    /// from Geometry to DecorativeGeometry.
+    void setDecorativeGeometryAppearance(
+        SimTK::DecorativeGeometry& decoration) const {
+        decoration.setColor(get_Appearance().get_color());
+        decoration.setOpacity(get_Appearance().get_opacity());
+        if (get_Appearance().get_visible())
+            decoration.setRepresentation(
+                (VisualRepresentation)
+                get_Appearance().get_representation());
```

Comment:
You are not setting the Appearance of DecorativeGeometry, right? This method translates the appearance of the OpenSim::Geometry to the attributes of the underlying SimTK::DecorativeGeometry. This distinction has to be more clear otherwise it is very confusing.

If the underlying `SimTK::DecorativeGeometry` only has a `representation` and you are are short-circuiting the `SurfaceAppearance` accessor with `Appearance::get_Appearance()`, why isn't `representation` just a member of Appearance? If the underlying `DecorativeGeometry` has not distinction, what is the utility of having it here? It seems above and below you just need a `representation`. 

How is (will) texture (be) handled by `DecorativeGeometry`? I ask because I don't see where there is any distinction made between the underlying `DecorativeGeometry` (which has to understand *representation* and *texture*?), which makes me wonder why is there is a separation of Surface and Curve properties inside Appearance if the `DecorativeGeometry` doesn't distinguish them?